patches/api/0324-Add-better-player-experience-API.patch

```diff
@@ -0,0 +1,28 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: jakebacker <jbacker42@gmail.com>
+Date: Sat, 24 Jul 2021 19:29:17 -0400
+Subject: [PATCH] Add better player experience API
+
+This patch adds Player#getCurrentExperience which returns the total experience point a player currently has.
+
+diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
+index 5e8814cc317a705eaf8bdd9f3876a5366c0a0226..fa9ff36bdb2713c8e446e4f9db27f394cf14ea9a 100644
+--- a/src/main/java/org/bukkit/entity/Player.java
++++ b/src/main/java/org/bukkit/entity/Player.java
+@@ -1178,6 +1178,16 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
+      */
+     public void setTotalExperience(int exp);
+ 
++    // Paper start
++    /**
++     * Gets the player's current experience points.
++     * <br>
++     * This refers to the total amount of experience the player <b>currently</b> has.
++     * @return Current experience points
++     */
++    public long getCurrentExperience();
```

Comment:
I think this method should be on the `HumanEntity` interface, ik most of the methods are on `Player` but there's on XP method on HumanEntity and that matches up with the experience fields being on nms Player not ServerPlayer